mm/slub.c

```diff
@@ -2583,6 +2583,20 @@ static inline void *get_freelist(struct kmem_cache *s, struct page *page)
 	return freelist;
 }
 
+/*
+ * If using KFENCE, allow passing object size in the top 16 bits of _RET_IP_,
+ * assuming they are set to 0xFFFF otherwise (currently only 64-bit systems).
+ */
+#ifdef CONFIG_KFENCE
+#define RET_IP_SIZE(size) \
+	((((unsigned long)(_RET_IP_) << 16) >> 16) | \
+	 ((unsigned long)size << 48))
```

Comment:
I think this needs to be calculated in terms of BITS_PER_LONG.

Reply:
Noted, but I'll be dropping this patch now.